### PR TITLE
chore: Split Docs CI from core CI

### PR DESCRIPTION
**Pull request by nzakas**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[x] Other, please explain:

CI update

#### What changes did you make? (Give an overview)

This is an attempt to split out the docs-related CI from everything else to help non-docs-related PRs finish their CI faster.

I've split out the docs-related CI tasks into a separate workflow that only runs when files are updated in the `docs/` directory.

#### Is there anything you'd like reviewers to focus on?

Did I miss anything?

### CODE REVIEW

**Review by snitin315 — APPROVED** [review]
LGTM. I will leave it open for others to review.

**Review by harish-sethuraman — APPROVED** [review]
LGTM

**Comment by harish-sethuraman** [comment]
Only doubt I have is we have added paths for the docs CI. Should we consider excluding the docs in the normal one then?
Any change inside docs directory only might trigger the main CI as well? Just wanted to confirm on this alone


Just in case:
```
 paths-ignore:
      - 'docs/**'
```

**Comment by mdjermanovic** [comment]
> Should we consider excluding the docs in the normal one then?

Some checks verify consistency between code and docs, so either change should trigger them and thus we'd need to split checks into more than 2 sets.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -19,28 +19,15 @@ jobs:
...
-    - name: Install Docs Packages
-      working-directory: docs
-      run: npm install
-    - name: Stylelint Docs
-      working-directory: docs
-      run: npm run lint:scss
-    - name: Lint Docs JS Files
-      run: node Makefile lintDocsJS
-    - name: Check Rule Examples
-      run: node Makefile checkRuleExamples
```
[the remark this answers is not in the document]
Comment (nzakas) [reply .github/workflows/ci.yml:37]: Makes sense. :+1:

**Comment by mdjermanovic** [comment]
There are merge conflicts now.

File: .github/workflows/docs-ci.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+name: CI
+on:
+  push:
+    branches: [main]
+    paths:
+        - 'docs/**'
+
+  pull_request:
+    branches: [main]
+
```
Comment (mdjermanovic) [inline .github/workflows/docs-ci.yml:10]: Shouldn't we also add the same `paths` for `pull_request`?

Comment (nzakas) [reply .github/workflows/docs-ci.yml:10]: Ooh good catch!

**Pull request merged**